### PR TITLE
Minor: change visibility of hash join utils

### PR DESCRIPTION
**Pull request by berkaysynnada**
## Which issue does this PR close?

Closes #.

## Rationale for this change

These functions can be used in other modules across the crate, so I'm relaxing their accessibility.

### CODE REVIEW

**Review by ozankabak — APPROVED** [review]
LGTM -- we indeed find these utility functions useful in our fork, which may be the case for others too. Instead of making everyone duplicate the logic, let's make them crate-accessible.

File: datafusion/core/src/physical_optimizer/join_selection.rs [hunk 1]
```diff
@@ -176,7 +176,7 @@ fn swap_join_projection(
 /// This function swaps the inputs of the given join operator.
 /// This function is public so other downstream projects can use it
 /// to construct `HashJoinExec` with right side as the build side.
-pub fn swap_hash_join(
+pub(crate) fn swap_hash_join(
```
Comment (alamb) [inline datafusion/core/src/physical_optimizer/join_selection.rs:179]: I believe this is actually a downgrade of visibility -- it used to be `pub` but now is `pub(crate)` meaning that comet can't use it: https://github.com/apache/datafusion/issues/13898

I will make a PR to fix this

Comment (ozankabak) [reply datafusion/core/src/physical_optimizer/join_selection.rs:179]: Oops, sorry for that. I guess we hastily thought `pub(crate)` would be sufficient but in Comet's case they need the full `pub`.

Comment (alamb) [reply datafusion/core/src/physical_optimizer/join_selection.rs:179]: No worries -- it was easily fixed

**Pull request merged**